src/stdlib_logger.f90

```diff
@@ -0,0 +1,1431 @@
+        logical, intent(out), optional              :: indent
+!! A logical flag to indent subsequent lines
+        integer, intent(out), optional              :: max_width
+!! The maximum number of columns for most outputs
+        logical, intent(out), optional              :: time_stamp
+!! A logical flag to add a time stamp
+        integer, intent(out), allocatable, optional :: log_units(:)
+!! The I/O units used in output
+
+!!##### Example
+!!
+!!     module example_mod
+!!       use stdlib_logger
+!!       ...
+!!     contains
+!!       ...
+!!       subroutine example_sub(unit, ...)
+!!         integer, intent(in) :: unit
+!!         ...
+!!         integer, allocatable :: log_units(:)
+!!         ...
+!!         call global_logger % configuration( log_units=log_units )
+!!         if ( size(log_units) == 0 ) then
+!!            call add_logger_unit( unit )
+!!         end if
+!!         ..
+!!       end subroutine example_sub
+!!       ...
+!!     end module example_mod
+
+
+        if ( present(add_line) ) add_line = self % add_line
+        if ( present(indent) ) indent = self % indent_lines
+        if ( present(max_width) ) max_width = self % max_width
+        if ( present(time_stamp) ) time_stamp = self % time_stamp
+        if ( present(log_units) ) log_units = self % log_units(1:self % units)
+
+    end subroutine configuration
+
+
+    pure subroutine configure( self, add_line, indent, max_width, time_stamp )
+!! Configures the logging process for SELF. The following attributes are
+!! configured:
+!! 1. ADD_LINE is a logical flag with .TRUE. implying that output starts
+!!    with a blank line, and .FALSE. implying no blank line. ADD_LINE has a
+!!    default value of .TRUE..
+!! 2. INDENT is a logical flag with .TRUE. implying that subsequent lines
+!!    will be indented 4 spaces and .FALSE. implying no indentation. INDENT
+!!    has a default value of .TRUE..
+!! 3. MAX_WIDTH is the maximum number of columns of output text with
+!!    MAX_WIDTH == 0 => no bounds on output width. MAX_WIDTH has a default
+!!    value of 80.
+!! 4. TIME_STAMP is a logical flag with .TRUE. implying that the output
+!!    will have a time stamp, and .FALSE. implying that there will be no
+!!    time stamp. TIME_STAMP has a default value of .TRUE..
+
+!!##### Example
+!!
+!!     program main
+!!         use stdlib_logger
+!!         ...
+!!         call global_logger % configure( indent=.false., max_width=72 )
+!!         ...
+
+        class(logger_t), intent(inout) :: self
+        logical, intent(in), optional  :: add_line
+        logical, intent(in), optional  :: indent
+        integer, intent(in), optional  :: max_width
+        logical, intent(in), optional  :: time_stamp
+
+        if ( present(add_line) ) self % add_line = add_line
+        if ( present(indent) ) self % indent_lines = indent
+        if ( present(max_width) ) then
+            if ( max_width <= 4 ) then
+                self % max_width = 0
```

Comment:
Should it be set to the default value 80, instead of 0?

Reply:
The idea i that setting `max_witdth` small indicates that formatting should be turned off.

Reply:
Sorry. I didn't read properly the specs: this is indeed mentioned.